lib/iris/_deprecation.py

```diff
@@ -33,6 +33,8 @@ class IrisDeprecation(UserWarning):
 
 
 def warn_deprecated(msg, **kwargs):
+    stacklevel = kwargs.get('stacklevel', 0) + 2
```

Comment:
A short docstring explaining that `stacklevel` is now relative to the calling frame would be the icing on the cake. Perhaps with a simple example?
```python
def warn_deprecated(msg, **kwargs):
    """
    $ cat << EOF > example.py
    from iris._deprecation import warn_deprecated

    def foo():
        warn_deprecated('Danger!')
        warn_deprecated('Awooga!', stacklevel=1)

    def bar():
        warn_deprecated('Evacuate!')
        foo()

    if __name__ == '__main__':
        bar()
    EOF
    $ python example.py
    example.py:8: IrisDeprecation: Evacuate!
      warn_deprecated('Evacuate!')
    example.py:4: IrisDeprecation: Danger!
      warn_deprecated('Danger!')
    example.py:9: IrisDeprecation: Awooga!
      foo()
    """
```